shuup/admin/modules/orders/views/status.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2016, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from shuup.admin.breadcrumbs import BreadcrumbedView
+from shuup.admin.utils.picotable import ChoicesFilter, Column, TextFilter
+from shuup.admin.utils.views import CreateOrUpdateView, PicotableListView
+from shuup.core.models import OrderStatus, OrderStatusManager, OrderStatusRole
+from shuup.utils.multilanguage_model_form import MultiLanguageModelForm
+
+
+class OrderStatusForm(MultiLanguageModelForm):
+    identifier = forms.CharField(
+        max_length=64, label=_("Identifier"), help_text=_("Internal identifier to help identify the status."))
+
+    class Meta:
+        model = OrderStatus
+        exclude = ["default"]
+
+    def __init__(self, **kwargs):
+        super(OrderStatusForm, self).__init__(**kwargs)
+        if self.instance.pk and OrderStatusManager().is_default(self.instance):
+            del self.fields['identifier']
+            del self.fields['role']
+            del self.fields['ordering']
+            del self.fields['is_active']
+
+        self.initial["identifier"] = self.instance.identifier
+
+    def clean(self):
+        if self.instance.pk and OrderStatusManager().is_default(self.instance):
+            data = self.cleaned_data
+            data["identifier"] = self.instance.identifier
+            data["role"] = self.instance.role
+            data["ordering"] = self.instance.ordering
+            data["is_active"] = self.instance.is_active
+            return data
+        return super(OrderStatusForm, self).clean()
+
+    def save(self, commit=True):
+        self.instance.identifier = self.cleaned_data["identifier"]
+        return super(OrderStatusForm, self).save(commit)
+
+
+class OrderStatusEditView(BreadcrumbedView, CreateOrUpdateView):
+    model = OrderStatus
+    form_class = OrderStatusForm
+    template_name = "shuup/admin/orders/status.jinja"
+    context_object_name = "status"
+    parent_name = _("Order Statuses")
+    parent_url = "shuup_admin:order_status.list"
+
+
+class OrderStatusListView(PicotableListView):
+    model = OrderStatus
+    default_columns = [
+        Column("identifier", _("Identifier"), linked=True, filter_config=TextFilter(operator="startswith")),
```

Comment:
I think it should, it's the identifier of the field and in this case it's something that matter and helps merchant to identify field even when (s)he has multiple custom statuses